### ISSUE TITLE
Secondary and tertiary buttons have no focus ring when not fully disabled

### ISSUE THREAD

**Issue by andrewhayward** [WordPress/gutenberg#56149]
## What problem does this address?

Buttons support the ability to retain focus while being disabled, via the `__experimentalIsFocusable` prop. However, when this is enabled on secondary and tertiary buttons, they do not have a focus ring. For example, the "Reset filters" button here is focused and only mostly disabled, but that is not visibly evident.

![A toolbar with a search field, two buttons and a dropdown toggle. One button (circled in red) is disabled and focused, but has no focus ring.](https://github.com/WordPress/gutenberg/assets/159848/ef7ea361-1643-47a5-9d9d-5033ab838675)

This is caused by an issue of specificity in the CSS, with `.is-secondary` and `.is-tertiary` overriding previously established focus styling for buttons using `box-shadow` and `outline`:

https://github.com/WordPress/gutenberg/blob/1bf654b6c7f794d598fc6980789185924168ee2f/packages/components/src/button/style.scss#L125-L134

https://github.com/WordPress/gutenberg/blob/1bf654b6c7f794d598fc6980789185924168ee2f/packages/components/src/button/style.scss#L38-L43

## What is your proposed solution?

The fix could be as straightforward as changing the scope or specificity of one of the other rules, to allow for the focus styling to propagate. This is probably most easily achieved by moving the `box-shadow` and `outline` declarations into a sub-block:

```diff
diff --git a/packages/components/src/button/style.scss b/packages/components/src/button/style.scss
index 03273056cf..4b11d9169a 100644
--- a/packages/components/src/button/style.scss
+++ b/packages/components/src/button/style.scss
@@ -129,8 +129,11 @@
                        background: transparent;
                        transform: none;
                        opacity: 1;
-                       box-shadow: none;
-                       outline: none;
+
+                       &:not(:focus) {
+                               box-shadow: none;
+                               outline: none;
+                       }
                }
        }
```

This will at least allow mostly-disabled buttons to pick up the focus styling applied to other controls:

![A toolbar with a search field, two buttons and a dropdown toggle. One button is disabled and focused, and has a focus ring.](https://github.com/WordPress/gutenberg/assets/159848/fd68f8c7-59a8-4bc4-87a0-2f5714bc4db3)

**Comment by Mamaduka** [comment]
This came up in #54722. cc @jameskoster, @jasmussen

**Comment by jasmussen** [comment]
Seems like a bug, yes.

**Comment by joedolson** [comment]
I feel like a disabled control should have a different focus outline, to provide visual affordance that even though you can focus the control, it's not available to use. 

This is a fairly ambiguous situation, accessibility-wise; combining the needs of focus visibility with the affordance of a disabled control. I think we should de-saturate the focus ring when disabled controls have focus; possibly even add an additional icon indicator on the control to help indicate that it's disabled. Setting focus on a control usually implies that you can use the control, so we need to provide strong signals visually that this is disabled.
